From c6cda0b55eead053701097be830cd4e791100613 Mon Sep 17 00:00:00 2001
Message-Id: <c6cda0b55eead053701097be830cd4e791100613.1426574757.git.chang-joon.lee@intel.com>
In-Reply-To: <eba22108096acc90a0516b200c616db90ae0cb5f.1426574757.git.chang-joon.lee@intel.com>
References: <eba22108096acc90a0516b200c616db90ae0cb5f.1426574757.git.chang-joon.lee@intel.com>
From: Gajanan Bhat <gajanan.bhat@intel.com>
Date: Thu, 12 Mar 2015 16:01:21 +0530
Subject: [PATCH 8/8] FOR_UPSTREAM [VPG]: drm/i915: Do not touch clock or
 power domains for NULL fb modeset

A mode set with NULL fb is issued after a dpms off, there is no need to touch the
power domains at that time and doing so will make the display fail to enter D0ix.

Issue: GMINL-6994
Change-Id: I9ab9b26d7cb651721ee51e73ff8798cba8dd3201
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
Signed-off-by: S, Deepak <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index d96e053..9f3168a 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -12198,8 +12198,10 @@ static int __intel_set_mode(struct drm_crtc *crtc,
 	 * update the the output configuration. */
 	intel_modeset_update_state(dev, prepare_pipes);
 
-	if (dev_priv->display.modeset_global_resources)
-		dev_priv->display.modeset_global_resources(dev);
+	if (fb != NULL) {
+		if (dev_priv->display.modeset_global_resources)
+			dev_priv->display.modeset_global_resources(dev);
+	}
 
 	/* DO it only once */
 	if (IS_VALLEYVIEW(dev))
-- 
1.7.9.5

